practice.rb

```diff
@@ -0,0 +1,66 @@
+def print_array(a)
+  print "["
+  l = a.length
+  a.each do |item|
+    print "#{item}"
+    if item != a[l-1]
+      print ", "
+    end
+  end
+  print "]"
+end
+
+print_array([4,5,6,4,2,123,42])
+
+def stretch(list)
+  print "["
+  l = (list.length) * 2
+  index = 0
+  list.each do |item|
+    if item % 2 == 0
+      print "#{item/2}, #{item/2}"
+    end
+    if item % 2 == 1
+      print "#{item/2 + 1}, #{item/2}"
+    end
+    index += 1
+    if index < l
+      print ", "
+    end
+  end
+  print "]"
+end
+
+stretch([4,5,6,4,2,123,42])
+
+def numUnique(list)
+  count = {}
+  list.each do |item|
+    if count[item] == nil
+      count[item] = 1
+    else
+      count[item] +=1
+    end
+  end
+  array = []
+  count.each do |k,v|
+    array << k
+  end
```

Comment:
But this... can't you just know the size/length of a hash? Why do we need an array to get that information?